### PR TITLE
Add the remaining classes and create the associations between them

### PR DESCRIPTION
**Pull request by farouqdaif1**
## What does this PR do?
This PR is raised to set up associations as in UML diagram.
> 
## Description of the task to be completed
* [x]  Create a class `Classroom `
* [x]  Create the `has-many/belongs-to` relationship between Classroom and Student.
* [x]  Create a class `Book`
* [x]  Create a class `Rental`
* [x]  Create the `many-to-many (also has-many-through) ` relationship between `Person` and `Book` using the intermediate class `Rental`

## How should it be tested
Clone the repository
 
```
 git clone https://github.com/farouqdaif1/OOP-school-library
```
 
 Navigate to the folder
 
```
 cd OOP-school-library
 ```

 Open in VS Code

 ```
 code .
 ```

 #### Thanks for reviewing my PR 😊

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_accessor :data
```
Comment (rloterh) [inline rental.rb]: - [ ] OPTIONAL
Kindly correct the typo: `data` should read as `date` instead.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (rloterh) [inline book.rb]: - [ ] We need both   _Getters_ and _Setters_ for _@rentals_ to be present, therefore kindly ensure to mark _rentals_ with _attr_accessor_, similar to:
```ruby
  attr_accessor :rentals

File: person.rb [hunk 3]
```diff
@@ -8,6 +8,7 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
```
Comment (rloterh) [inline person.rb:11]: - [ ] We need both   _Getters_ and _Setters_ for _@rentals_ to be present, therefore kindly ensure to mark _rentals_ with _attr_accessor_, similar to:
```ruby
  attr_accessor :rentals

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Farouq, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Good Points 👍  
- Descriptive Pull Request
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly check the comments marked **[OPTIONAL]** under the review._
<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
Hi @farouqdaif1, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**